framework/aster-frame/src/console.rs

```diff
@@ -2,7 +2,12 @@
 
 use core::fmt::Arguments;
 
+use crate::sync::SpinLock;
+
+pub static PRINT_LOCK: SpinLock<()> = SpinLock::new(());
+
 pub fn print(args: Arguments) {
+    let guard = PRINT_LOCK.lock_irq_disabled();
```

Comment:
Why we need this `PRINT_LOCK`? I scan the code and find virtio-console and serial port share the same lock, but they are independent, right?

Reply:
The PRINT_LOCK is implemented to guarantee atomicity when outputting a complete message. This is because write_fmt eventually outputs in chunks by making multiple calls to write_str. Merely relying on the atomicity provided by the console device is insufficient. It is correct to use separate locks for the virtio-console and the serial port.